### ISSUE TITLE
Runtime Version Check for Cuda and OpenCL

### ISSUE THREAD

**Issue by NicolasDenoyelle** [anlsys/aml#180]
* Cuda version is not checked in configure.
* Cuda and OpenCL version need to be checked at runtime -- when checking backend support -- so that headers match AML linked libraries version.

**Comment by perarnau** [comment]
Yes. We might need to pay attention to semver on those checks, as in theory the headers & lib just need to be ABI compatible, not equal. If I recall properly it means major-version matches only.
